hashintel/hash
# H-480: Scaffold `hashdotai-new` app, upgrade Next and MUI

## 🌟 What is the purpose of this PR?

We will be incrementally migrating the site at `https://hash.ai` from a legacy private codebase to this repo.

This PR adds the scaffold for a new NextJS site, at `apps/hashdotai-new`. We will build new pages here and serve them under specific `hash.ai/xxx` routes as they become available. This site uses the new(ish) NextJS App Router approach. 

This PR also takes the opportunity to upgrade Next and MUI everywhere, since we're fairly behind on Next, and there are various fixes to do with the interplay between each and the new App router/Server Components we can benefit from.

## Pre-Merge Checklist 🚀

### 🚢 Has this modified a publishable library?




This PR:



### 📜 Does this require a change to the docs?




The changes in this PR:


### 🕸️ Does this require a change to the Turbo Graph?




The changes in this PR:


## ⚠️ Known issues

- NextJS has a bunch of build warnings related to MUI, described fully in the new `README.md` in `apps/hashdotai-new`. If these become particularly annoying we can downgrade NextJS to `15.4.x` until they are fixed, but it would be nice to have the DX benefits of `15.5.x` ([various speed increases promised](https://nextjs.org/blog/next-13-5)).

## 🐾 Next steps

- Once this is merged, add a Vercel project for `hashdotai-new`

## 🛡 What tests cover this?



- Some basic ones for `hash-frontend`, none otherwise.

## ❓ How to test this?



1. Check that `hashdotdev` and `hash-frontend` still work (can be done on the deployment)
2. Check that `yarn dev` works locally for the new package
